libraries/AP_Scripting/generator/src/main.c

```diff
@@ -615,6 +615,7 @@ int parse_type(struct type *type, const uint32_t restrictions, enum range_check_
       case TYPE_FLOAT:
       case TYPE_INT32_T:
       case TYPE_UINT32_T:
+      case TYPE_USERDATA:
```

Comment:
What is the intent of this?

Reply:
we create the cylinder status as a separate object then need to assign it in EFI_State. It is a structure, so can't be represented numerically, so makes no sense to do range checking. If we don't make this change we don't compile

Reply:
I agree that is doesn't build, but skip check on userdata doesn't do anything because there is nothing to check. So if you drop this commit and change:
``` 
userdata EFI_State field cylinder_status Cylinder_Status'skip_check write
```
to
```
userdata EFI_State field cylinder_status Cylinder_Status write
```
The generated bindings are identical.